base/linalg/bidiag.jl

```diff
@@ -271,10 +273,11 @@ function diag(M::Bidiagonal{T}, n::Integer=0) where T
         return M.uplo == 'U' ? M.ev : zeros(T, size(M,1)-1)
     elseif n == -1
         return M.uplo == 'L' ? M.ev : zeros(T, size(M,1)-1)
-    elseif -size(M,1) < n < size(M,1)
+    elseif -size(M,1) <= n <= size(M,1)
```

Comment:
Is the upper bound meant to be size(M,2) here?

Reply:
Or possibly size(M,1) instead of 2 in the exception below

Reply:
`M` is square so it doesn't matter